File: CS214-Programming Language Concepts/labs/04/C++/log_table.cpp
/* log_table.cpp displays a table of logarithms.
 *
 * Input: start, stop and increment, three doubles.
 * PRE: start < stop && increment > 0.
 * Output: A table of logarithms from start to stop,
 *         with increment as the step value.
 * Begun by: Adams, for CS 214 at Calvin College.
 * Completed by:Joshua Maguire
 ********************************************************/

#include <iostream>    // cin, cout, <<, >>
#include <cassert>     // assert()
#include <cmath>       // log10()
using namespace std;

int main() 
{
  cout << "\nTo display a table of logarithms,"
       << "\n enter the start, stop and increment values: ";
  double start, stop, increment;
  cin >> start >> stop >> increment;
  
  assert(increment > 0);
  
   for (double count = start; count <= stop; count += increment)
      cout << "\nThe logarithm of " << count
           << " is " << log10(count) << endl;
}

File: CS214-Programming Language Concepts/labs/04/ruby/log_table.rb
#! /usr/bin/ruby
# log_table.rb displays a table of logarithms
#
# Begun by: Dr. Nelesen, for CS 214 at Calvin College.
# Completed by:Joshua Maguire
#
# Input:  The start, stop and increment values
# Precondition: The start value is less than the stop value, 
#     and the increment is greater than 0
# Output: A table of logarithms from start to stop, with increment
#     as the step value


if __FILE__ == $0
   print "Enter the start value: "
	 start = gets.chomp.to_i #Gets int from keyboard
   print "Enter the stop value: "
	 stop = gets.chomp.to_i #Gets int from keyboard
   print "Enter the increment value: "
	 counter = gets.chomp.to_f #Gets int from keyboard
	 
	 
	 while start <= stop #our simple loop
		puts "the logarithm of #{start} is #{Math.log10(start)}" #use #{value}, (rubys string literal thing
		start = start + counter #increment counter
	 end
end

